### ISSUE TITLE
Configuration for JavaMailSender has to be repeated

### ISSUE THREAD

**Issue by oguera** [spring-projects/spring-boot#18958]
## Version
Spring Boot: v2.1.1.RELEASE
spring-boot-start-mail: 2.2.1.RELEASE

## Context
See this example application.yml:

```
spring:
  mail:
    host: mail.my-domain.tld
    username: myusername
    port: 587
    password: mypassword
    test-connection: true
    tls: true
    properties:
      mail:
        debug: true
        transport:
          protocol: smtp
        smpt:
          port: 587
          host: mail.my-domain.tld
          auth: true
          starttls:
            enable: true
```

Using this and only this config (without manual bean creation) i'll get this exception:
`mail com.sun.mail.smtp.SMTPSendFailedException: 530 5.7.0 Must issue a STARTTLS command first`

This issue is only resolved after i create the bean myself and just repeat the configuration from obove but this time in Java:

```
@Bean
    public JavaMailSender getJavaMailSender() {
        JavaMailSenderImpl mailSender = new JavaMailSenderImpl();
        mailSender.setHost(hostFromConfig);
        mailSender.setPort(587);

        mailSender.setUsername(mailFromConfig);
        mailSender.setPassword(passwordFromConfig);

        Properties props = mailSender.getJavaMailProperties();
        props.put("mail.transport.protocol", "smtp");
        props.put("mail.smtp.auth", "true");
        props.put("mail.smtp.starttls.enable", "true");
        props.put("mail.debug", "true");

        return mailSender;
    }
```

After adding this snippet i can send mails.

Shouldn't the Autoconfiguration for the `JavaMailSender` construct the Bean just like i did manually with the `application.yml` File?

## What i did so far
[This should be the relevant section for creating the Bean ](https://github.com/spring-projects/spring-boot/blob/b9f4a9c075102ef04956139079a4e25bddd989e8/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/mail/MailSenderPropertiesConfiguration.java#L49)

**Comment by dreis2211** [comment]
Hi @oguera . Not a team-member, but your configuration seems to contain a typo. It should be probably `smtp` instead of `smpt`

**Comment by oguera** [comment]
Nice catch @dreis2211. I checked everything for a typo a few times but it seems i didn't see the forest for the trees after a while. 

I fixed the typo in the `application.yml` and it works as expected. Strange thing (not regarding spring itself) i came up with that approach because i read it in [this article on baeldung](https://www.baeldung.com/spring-email) where it seems that they are also doing the configuration twice. 

Again thank you for pointing it out. I'll close the issue.
